src/Google.Api.Gax/PageStreamer.cs

```diff
@@ -91,25 +98,35 @@ public IEnumerable<TResource> Fetch(TRequest initialRequest, Func<TRequest, TRes
         /// <summary>
         /// Lazily and asynchronously fetches resources a page at a time.
         /// </summary>
+        /// <param name="callSettings">The <see cref="CallSettings"/> to apply to each RPC invocation.</param>
         /// <param name="initialRequest">The initial request to send. If this contains a page token,
         /// that token is maintained. This is also passed to the request provider along with a token
         /// to provide another request. (The provider may build a new object, or modify the existing request.)</param>
         /// <param name="responseFetcher">A function to asynchronously fetch a response containing a page of
         /// results given a request. Must not be null</param>
         /// <returns>A sequence of resources, which are fetched asynchronously and a page at a time.</returns>
-        public IAsyncEnumerable<TResource> FetchAsync(TRequest initialRequest, Func<TRequest, CancellationToken, Task<TResponse>> responseFetcher)
+        public IAsyncEnumerable<TResource> FetchAsync(
+            CallSettings callSettings,
+            TRequest initialRequest,
+            ApiCall<TRequest, TResponse>.AsyncCall responseFetcher)
         {
             GaxPreconditions.CheckNotNull(responseFetcher, nameof(responseFetcher));
             TRequest nextRequest = initialRequest;
             bool done = false;
+            CallSettings rpcCallSettings = callSettings?.Clone() ?? new CallSettings();
             return new PageStreamingAsyncEnumerable<TResource>(async cancellationToken =>
             {
-                cancellationToken.ThrowIfCancellationRequested();
+                // If MoveNext() has been passed a non-default cancellation-token, then use it.
+                // Otherwise, use the callSettings cancellation-token (if it has one).
+                CancellationToken? rpcCancellationToken = cancellationToken != default(CancellationToken) ?
```

Comment:
I suspect we want this check elsewhere, too - we might want to add extension methods to make it simpler. Not yet though.